actions/actions.py

```diff
@@ -9,27 +9,45 @@
 
 from rasa_sdk import Action, Tracker
 from rasa_sdk.executor import CollectingDispatcher
-import requests
+import aiohttp
+import json
+import logging
 
+logger = logging.getLogger(__name__)
 
 class ActionHaystack(Action):
 
     def name(self) -> Text:
         return "call_haystack"
 
-    def run(self, dispatcher: CollectingDispatcher,
+    async def run(self, dispatcher: CollectingDispatcher,
             tracker: Tracker,
             domain: Dict[Text, Any]) -> List[Dict[Text, Any]]:
 
         url = "http://localhost:8000/query"
-        payload = {"query": str(tracker.latest_message["text"])}
+        payload = {
+            "query": str(tracker.latest_message["text"]),
+            "params":{"filters": {}, "Retriever": {"top_k": 3}, "Reader": {"top_k": 3}, "Query": {"debug": False}}
+        }
         headers = {
             'Content-Type': 'application/json'
         }
-        response = requests.request("POST", url, headers=headers, json=payload).json()
+
+        logger.debug(f"Calling {url}, ask: {str(tracker.latest_message['text'])}")
+        try:
+            async with aiohttp.ClientSession() as session:
+                async with session.post(url, headers=headers, json=payload) as response:
+                    response = await response.json()
+        except aiohttp.ClientConnectorError as e:
+            logger.error(f"Haystack service not responding: {str(e)}")
+            response = {"answers":[{"answer":f"Haystack service not responding: {str(e)}"}]}
 
         if response["answers"]:
+            logger.debug(f"Answers returned by haystack:\n{response}")
             answer = response["answers"][0]["answer"]
+            if not answer:
+                answer = response["answers"][1]["answer"]
+            logger.debug(f"selected answer: {answer}")
         else:
             answer = "No Answer Found!"
```

Comment:
the `answer = response["answers"][1]["answer"]` might cause an index error if there are no answers at all or less than 2 answers
```suggestion
        answer = "No Answer Found!"
        for answer_object in response.get("answers") or []:
            current_answer = answer_object["answer"]
            if current_answer:
                answer = current_answer
                logger.debug(f"selected answer: {answer}")
                break
```